### PR TITLE
#503a - Update SkuPriceId per SkuPriceDetails

### PR DESCRIPTION
**Pull request by kk09v**
This change is dependent on #503 being merged as it links to SkuPriceDetails. This change to SkuPriceId solves for a case where a provider does not have a SkuPriceId but wants to include SkuPriceDetails. This case was introduced when there was a change in scope of #503 to not include SkuDetails.

Changes:
- Reformatted normative to bullets
- Added option (MAY) to repeat SkuId as SkuPriceId when there is no SkuPriceId, as long as other conditions are not violated.

### CODE REVIEW

**Comment by jpradocueva** [comment]
### Action Items from Maintainers call on Aug 26th:
[Maintainers-[#530](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/530) ] Karl - @kk09v : To refine the PR and prepare it for Task Force Two’s discussion.

File: specification/columns/skupriceid.md [hunk 1]
```diff
@@ -2,7 +2,14 @@
...
 
-The SkuPriceId column MUST be present in a FOCUS dataset when the provider publishes a SKU price list. This column MUST be of type String. SkuPriceId MUST define a single unit price used for calculating the charge. The [ListUnitPrice](#listunitprice) MUST be associated with the SkuPriceId in the provider published *price list*. This column MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory. A given value of SkuPriceId MUST be associated with one and only one [SkuId](#skuid), except in cases of commitment discount flexibility.
+- The SkuPriceId column MUST be present in a FOCUS dataset when the provider publishes a SKU price list.
+- This column MUST be of type String.
```
Comment (udam-f2) [inline specification/columns/skupriceid.md:6]:
```suggestion
The SkuPriceId column adheres to the following requirements:
- The SkuPriceId MUST be present in a FOCUS dataset when the provider publishes a SKU price list and MUST be of type String.
```

File: specification/columns/skupriceid.md [hunk 2]
```diff
@@ -2,7 +2,14 @@
...
+- SkuPriceId MUST define a single unit price used for calculating the charge.
+- The [ListUnitPrice](#listunitprice) MUST be associated with the SkuPriceId in the provider published *price list*.
```
Comment (udam-f2) [inline specification/columns/skupriceid.md:8]:
```suggestion
- [ListUnitPrice](#listunitprice) MUST be associated with the SkuPriceId in the provider published *price list*.
```

File: specification/columns/skupriceid.md [hunk 3]
```diff
@@ -2,7 +2,14 @@
...
+- This column
+- MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory.
```
Comment (udam-f2) [inline specification/columns/skupriceid.md:10]:
```suggestion
- SKUPriceId MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory.
```

**Review by flanakin — APPROVED** [review]
I would like to request that we not mix reformatting and edits in the same PR. That makes it hard to spot what changed, whether intentional or accidental.

File: specification/columns/skupriceid.md [hunk 4]
```diff
@@ -2,7 +2,13 @@
...
 
-The SkuPriceId column MUST be present in a FOCUS dataset when the provider publishes a SKU price list. This column MUST be of type String. SkuPriceId MUST define a single unit price used for calculating the charge. The [ListUnitPrice](#listunitprice) MUST be associated with the SkuPriceId in the provider published *price list*. This column MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory. A given value of SkuPriceId MUST be associated with one and only one [SkuId](#skuid), except in cases of commitment discount flexibility.
+The SkuPriceId column adheres to the following requirements:
+- The SkuPriceId MUST be present in a FOCUS dataset when the provider publishes a SKU price list and MUST be of type String.
```
Comment (kk09v) [inline specification/columns/skupriceid.md:6]:
```suggestion
- SkuPriceId MUST be present in a FOCUS dataset when the provider publishes a SKU price list and MUST be of type String.
```

**Review by udam-f2 — APPROVED** [review]

**Review by ljadvey — APPROVED** [review]

**Comment by jpradocueva** [comment]
### Action Items from TF-2 call on Aug 28:
- [ ] [TF2-[#530](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/530)] Karl - @kk09v : To resolve the pending comment in PR #530 and commit any necessary changes.
- [ ] [TF2-[#530](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/530)] Karl  - @kk09v: To standardize bullet point formatting across the document.
- [ ] [TF2-[#530](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/530)] Shawn - @salpaysenturus and Karl  - @kk09v: To discuss offline the potential changes related to the use of SKU Price ID in edge cases.

**Review by ijurica — APPROVED** [review]

**Review by ahullah — APPROVED** [review]

**Comment by jpradocueva** [comment]
Approved by the Members group on Sep 5 call.

**Pull request merged**